R/model.R

```diff
@@ -664,8 +664,13 @@ process_data <- function(data) {
   } else if (is.character(data)) {
     path <- absolute_path(data)
   } else if (is.list(data) && !is.data.frame(data)) {
-    path <- tempfile(pattern = "standata-", fileext = ".dat")
-    rstan::stan_rdump(names(data), file = path, env = list2env(data))
+    if (cmdstan_version() >= "2.22") {
+      path <- tempfile(pattern = "standata-", fileext = ".json")
+      write_stan_json(data = data, file = path)
+    } else {
+      path <- tempfile(pattern = "standata-", fileext = ".dat")
```

Comment:
Before calling `rstan::stan_rdump` we should check that rstan can be loaded. Can you add something like

```r
if (!requireNamespace("rstan", quietly = TRUE)) {
  stop("For CmdStan < 2.22 the rstan package is required for writing data.")
}
```
or some other message indicating rstan needs to be installed?

Reply:
Good call. Thanks